packages/tenant-readmodel-writer/src/tenantConsumerService.ts

```diff
@@ -50,11 +53,67 @@ export async function handleMessage(
           }
         )
     )
-    .with({ type: "SelfcareMappingCreated" }, async (_msg) => {
-      logger.info("TODO");
+    .with({ type: "SelfcareMappingCreated" }, async (msg) => {
+      await tenants.updateOne(
+        {
+          "data.id": msg.stream_id,
+          "metadata.version": { $lt: msg.version },
+        },
+        {
+          $set: {
+            "data.selfcareId": msg.data.selfcareId,
+            "metadata.version": msg.version,
+          },
+        }
```

Comment:
```suggestion
        },
        { upsert: true }
```
Shouldn't we add this too?

Reply:
Probably it's not needed. As far as I understand, this event means we already have a tenant in reamodel, and we are setting its selfcareId to complete the onboarding. Let's double-check with @galales

Reply:
in this case the `upsert` would create an object with just `id` and `selfcareId`.
I see that for other services, v1 events that do no contain the whole object do not `upsert`, let's use the same approach.

Note: for v2 events is different, we should always upsert because we may use Kafka compaction